datafusion/sqllogictest/test_files/array.slt

```diff
@@ -1752,6 +1752,101 @@ select array_to_string(make_array(), ',')
 ----
 (empty)
 
+
+## array_union (aliases: `list_union`)
+
+# array_union scalar function #1
+query ?
+select array_union([1, 2, 3, 4], [5, 6, 3, 4]);
+----
+[1, 2, 3, 4, 5, 6]
+
+# array_union scalar function #2
+query ?
+select array_union([1, 2, 3, 4], [5, 6, 7, 8]);
+----
+[1, 2, 3, 4, 5, 6, 7, 8]
+
+# array_union scalar function #3
+query ?
+select array_union([1,2,3], []);
+----
+[1, 2, 3]
+
+# array_union scalar function #4
+query ?
+select array_union([1, 2, 3, 4], [5, 4]);
+----
+[1, 2, 3, 4, 5]
+
+# array_union scalar function #5
+statement ok
+CREATE TABLE arrays_with_repeating_elements_for_union
+AS VALUES
+  ([1], [2]),
+  ([2, 3], [3]),
+  ([3], [3, 4])
+;
+
+query ?
+select array_union(column1, column2) from arrays_with_repeating_elements_for_union;
+----
+[1, 2]
+[2, 3]
+[3, 4]
+
+statement ok
+drop table arrays_with_repeating_elements_for_union;
+
+# array_union scalar function #6
+query ?
+select array_union([], []);
```

Comment:
checked in spark 
```
scala> spark.sql("select array_union(array(), array())").show(false)
+-----------------------------+                                                 
|array_union(array(), array())|
+-----------------------------+
|[]                           |
+-----------------------------+
```

Reply:
I am even surprised my code work here tbh, do I need to add an additional branch to pattern matching where both array have null data type and return an empty array?

Reply:
> I am even surprised my code work here tbh, do I need to add an additional branch to pattern matching where both array have null data type and return an empty array?

I prefer return empty array in this case

Reply:
> I am even surprised my code work here tbh, do I need to add an additional branch to pattern matching where both array have null data type and return an empty array?

I don't think you should check the input types (they should still be lists, they'll just have an offset buffer of `[0, 1]`

Maybe you just need to handle the case specially in `array_union`